### PR TITLE
fix: issue #3

### PR DESCRIPTION
**Pull request by Veenoway**
- Role Priority Array: An array that ranks roles from the least important to the most important.
- Role-Based Content Functionality: A function that displays content depending on user roles.

if userIsPartOfMonad 
     find the most important role and return it;
else 
     return the message "You're not a Nad";

### CODE REVIEW

File: src/constants/monad_roles.ts [hunk 1]
```diff
@@ -15,3 +15,14 @@ export const MONAD_ROLES = {
 };
 
 export const MONAD_DEFAULT_ROLE_NAME = 'pleb';
+
+export type Role = 'nads' | 'full_access' | 'nad-og' | 'running_hot' | 'mon' | 'mon2' | 'monartist';
+export const rolePriority: Role[] = [
```
Comment (portdeveloper) [inline src/constants/monad_roles.ts:20]: This is good! But I believe that ranking roles might be a little bit of overdoing it as these roles do not have clear rankings.

So maybe what we should do here is to not differentiate between any role, and just show all the roles or at least full access.

If we are still sure to implement a ranking, we may want to do that in another PR 🙏

File: src/app/components/Home.tsx [hunk 2]
```diff
@@ -1,31 +1,42 @@
...
+
+  // Retrieves a personalized message based on the user's roles, if the user is part of the Monad group.
+  const getContentFromRole = () => {
+    if (userIsPartOfMonad) {
+      const role = rolePriority.find((role) => roleNames.includes(role));
+      if (role) {
+        return `You're all set, looks like you're indeed a ${role}`;
+      }
+    }
+    return "Oops, looks like you're not a NAD yet :(";
```
Comment (portdeveloper) [inline src/app/components/Home.tsx:36]: Maybe we should not be saying not a NAD if the user is not a member

**Review by portdeveloper — CHANGES_REQUESTED** [review]
Thanks for the good work @Veenoway !

**Comment by portdeveloper** [comment]
Hey thanks for the changes!

Two little nitpicks from me:
Home.tsx line 33: (capitalization)
      `else if (roleNames.includes('full_access')) return `${successText} Full access`;`
should be 
      `else if (roleNames.includes('full_access')) return `${successText} full access`;`
      
      and
      
Home.tsx line 34: (pluralization)
      `else return `Oh! Looks like you're still a newbies`;`
should be 
      `else return `Oh! Looks like you're still a newbie`;`

**Review by portdeveloper — APPROVED** [review]
Thanks for the good work @Veenoway !

And sorry for being this nitpicky haha.

**Comment by Veenoway** [comment]
> Thanks for the good work @Veenoway !
> 
> And sorry for being this nitpicky haha.

No Problem about it it's normal! 
Thank to let us the opportunity to help

**Pull request merged**